### PR TITLE
fix(#2890): Include retest requests in In Progress dashboard widget

### LINKED ISSUE #2890: Requesting for a retest whould show up in the In Progress widget of the home dashboard
When a user asks to retest an order under validation, The order should show up in the **In Progress** widget.

### PR DESCRIPTION
**Pull request by princegarg001**
## Summary

This PR addresses Issue #2890 where retest requests were not appearing in the "In Progress" widget on the home dashboard, causing laboratory staff to miss samples that required retesting.

### Problem Statement
When a laboratory technician or biologist rejects a sample result and requests a retest, the sample's analysis status changes to `BiologistRejected` or `TechnicalRejected`. However, the home dashboard's "In Progress" widget was only querying for `NotStarted` status, causing these retest requests to be invisible to lab staff.

### Root Cause Analysis
The `PatientDashBoardProvider.java` controller was using `getAnalysesForStatusId()` with only `AnalysisStatus.NotStarted`, excluding samples that were rejected and awaiting retest.

### Solution Implemented

**1. Added New DAO Method:**
- Created `getAnalysesForStatusIds(List<Integer> statusIdList)` method in `AnalysisDAO` and `AnalysisDAOImpl`
- This method accepts multiple status IDs and returns all matching analyses using HQL query

**2. Added New Service Method:**
- Extended `AnalysisService` interface with `getAnalysesForStatusIds()` method
- Implemented in `AnalysisServiceImpl` following the 5-layer architecture pattern

**3. Updated Dashboard Provider:**
- Modified `PatientDashBoardProvider.java` to include three statuses for "In Progress":
  - `AnalysisStatus.NotStarted` (original)
  - `AnalysisStatus.BiologistRejected` (new - retest requested by biologist)
  - `AnalysisStatus.TechnicalRejected` (new - retest requested by technician)
- Updated both the metrics count and list retrieval endpoints

**4. Added Comprehensive Tests:**
- Added unit tests in `AnalysisServiceTest.java`:
  - `getAnalysesForStatusIds_shouldReturnAnalysesWithMultipleStatuses()`
  - `getAnalysesForStatusIds_shouldReturnEmptyListForEmptyStatusList()`
- Added E2E tests in `home.cy.js` to verify API endpoints

### Files Modified

| File | Change Type |
|------|-------------|
| `src/main/java/org/openelisglobal/analysis/dao/AnalysisDAO.java` | Added method signature |
| `src/main/java/org/openelisglobal/analysis/daoimpl/AnalysisDAOImpl.java` | Implemented HQL query |
| `src/main/java/org/openelisglobal/analysis/service/AnalysisService.java` | Added service method |
| `src/main/java/org/openelisglobal/analysis/service/AnalysisServiceImpl.java` | Implemented service |
| `src/main/java/org/openelisglobal/common/rest/provider/PatientDashBoardProvider.java` | Updated queries |
| `src/test/java/org/openelisglobal/analysis/AnalysisServiceTest.java` | Added 2 unit tests |
| `frontend/cypress/e2e/home.cy.js` | Added E2E tests |

### Impact
- **Before:** Samples awaiting retest were not visible in the dashboard
- **After:** All samples requiring attention (new + retest) appear in the "In Progress" widget

## Screenshots

*This is a backend logic change. The UI remains the same, but the "In Progress" count and list now includes samples with retest status (`BiologistRejected`/`TechnicalRejected`).*
Screenshot 2026-03-08 171006
Screenshot 2026-03-08 171211

## Related Issue

Fixes #2890 - Requesting for a retest should show up in the In Progress widget of the home dashboard

### Testing Verification
- [x] Unit tests pass: `mvn test -Dtest=AnalysisServiceTest`
- [x] Code formatted with Spotless: `mvn spotless:apply`
- [x] Code formatted with Prettier: `npm run format`
- [x] Build compiles successfully: `mvn compile`
- [x] Follows 5-layer architecture pattern (DAO → Service → Controller)
- [x] No breaking changes to existing functionality

### Architecture Compliance
This implementation follows the OpenELIS Global constitution principles:
- ✅ **Principle IV (Layered Architecture):** Changes properly flow through DAO → Service → Controller
- ✅ **Principle V (Test-Driven Development):** Unit and E2E tests included
- ✅ **No hardcoded strings or UI changes**

---

Thank you for taking the time to review this PR! 🙏

I would greatly appreciate your feedback and verification. Please let me know if:
- Any additional tests are needed
- The implementation approach needs adjustment
- Any edge cases should be considered

Looking forward to your review and suggestions. Happy to make any changes as requested!

### CODE REVIEW

File: frontend/cypress/e2e/home.cy.js [hunk 1]
```diff
@@ -74,3 +74,35 @@ describe("User navigates to different tiles", function () {
     home.selectDelayedTurnAround();
   });
 });
+
+describe("In Progress widget includes retest requests (Issue #2890)", function () {
```
Comment (Vijaykv5) [inline frontend/cypress/e2e/home.cy.js:78]: Remove the unnecessary comments from the test please

Comment (princegarg001) [reply frontend/cypress/e2e/home.cy.js:78]: Sure I willl Do it

**Comment by mozzy11** [comment]
Thank you for this contribution, and apologies that it sat unreviewed for so long.

We are doing a cleanup of the PR queue: this PR is now more than 3 months old and has unresolved merge conflicts with develop. The codebase has changed substantially in that time (the frontend moved to Vite and is being migrated to TypeScript, the test infrastructure was rewritten, and several modules — test catalog, analyzers, EQA, storage — were reworked), so a mechanical rebase is unlikely to succeed.

Closing as stale per our queue policy. **If the problem you fixed still exists on current develop, please open a fresh PR against the latest code — we are actively reviewing now and it will get prompt attention.** Thanks again for contributing to OpenELIS!
